Maple2.Server.DebugGame/Graphics/DebugGraphicsContext.cs

```diff
@@ -240,66 +270,282 @@ private void OnLoad() {
             DxSwapChain = swapChain;
         }
 
-        if (VertexShader is null) {
-            VertexShader = new VertexShader(this);
-        }
+        // Create shaders
+        VertexShader ??= new VertexShader(this);
 
-        if (PixelShader is null) {
-            PixelShader = new PixelShader(this);
-        }
+        PixelShader ??= new PixelShader(this);
 
-        VertexShader.Load("screenVertex.hlsl", "vs_main");
-        PixelShader.Load("screenPixel.hlsl", "ps_main");
+        WireframeVertexShader ??= new VertexShader(this);
+
+        WireframePixelShader ??= new PixelShader(this);
+
+        VertexShader.Load(Path.Combine(Shader.ShaderRootPath, "screenVertex.hlsl"), "vs_main");
+        PixelShader.Load(Path.Combine(Shader.ShaderRootPath, "screenPixel.hlsl"), "ps_main");
+        WireframeVertexShader.Load(Path.Combine(Shader.ShaderRootPath, "wireframeVertex.hlsl"), "vs_main");
+        WireframePixelShader.Load(Path.Combine(Shader.ShaderRootPath, "wireframePixel.hlsl"), "ps_main");
+
```

Comment:
Should probably use the ShaderPipelines stuff for this. Also I'm seeing a lot of messy DirectX code littered everywhere. I think I'll do a pass of my own after the merge to clean it all up cause thats a lot to clean up and untangle that can't and shouldn't be vibe coded